src/transaction.cpp

```diff
@@ -1443,6 +1443,21 @@ int wsrep::transaction::certify_fragment(
            state() == s_must_abort);
 
     client_service_.wait_for_replayers(lock);
+
```

Comment:
Would it be possible to move this block below, after changing state to certifying? This way we would avoid extra unlock/lock and reduce possible races.

Reply:
done